ui/StatusQ/src/theme.cpp

```diff
@@ -0,0 +1,341 @@
+#include "StatusQ/theme.h"
+
+#include <QQmlApplicationEngine>
+#include <QQmlEngine>
+
+namespace {
+
+constexpr qreal s_defaultPadding = 16;
+constexpr qreal s_xlPaddingFactor = 2.0;
+constexpr qreal s_bigPaddingFactor = 1.5;
+constexpr qreal s_halfPaddingFactor = 0.5;
+constexpr qreal s_smallPaddingFactor = 0.625;
+
+constexpr int s_secondaryAdditionalTextBaseSize = 17;
+constexpr int s_primaryTextFontBaseSize = 15;
+constexpr int s_secondaryTextFontBaseSize = 14;
+constexpr int s_additionalTextBaseSize = 13;
+constexpr int s_tertiaryTextFontBaseSize = 12;
+constexpr int s_asideTextFontBaseSize = 10;
+
+const std::unique_ptr<ThemePalette> s_paletteDark = createDarkThemePalette();
+const std::unique_ptr<ThemePalette> s_paletteLight = createLightThemePalette();
+
+} // unnamed namespace
+
+Theme::Theme(QObject *parent)
+    : QQuickAttachedPropertyPropagator(parent), m_padding(s_defaultPadding)
+{
+    initialize();
+}
+
+qreal Theme::defaultPadding() const {
+    return s_defaultPadding;
+}
+
+qreal Theme::defaultXlPadding() const {
+    return s_defaultPadding * s_xlPaddingFactor;
+}
+
+qreal Theme::defaultBigPadding() const {
+    return s_defaultPadding * s_bigPaddingFactor;
+}
+
+qreal Theme::defaultHalfPadding() const {
+    return s_defaultPadding * s_halfPaddingFactor;
+}
+
+qreal Theme::defaultSmallPadding() const {
+    return s_defaultPadding * s_smallPaddingFactor;
+}
+
+qreal Theme::padding() const {
+    return m_padding;
+}
+
+qreal Theme::xlPadding() const {
+    return m_padding * s_xlPaddingFactor;
+}
+
+qreal Theme::bigPadding() const {
+    return m_padding * s_bigPaddingFactor;
+}
+
+qreal Theme::halfPadding() const {
+    return m_padding * s_halfPaddingFactor;
+}
+
+qreal Theme::smallPadding() const {
+    return m_padding * s_smallPaddingFactor;
+}
+
+qreal Theme::radius() const {
+    return s_defaultPadding * s_halfPaddingFactor;
+}
+
+void Theme::setPadding(qreal padding)
+{
+    auto explicitPaddingOld = m_explicitPadding;
+    m_explicitPadding = true;
+
+    if (m_padding == padding) {
```

Comment:
Uhm, comparing two floats with `==`? :) You should use `qFuzzyCompare` instead imo